packages/terra-date-picker/src/DateInput.jsx

```diff
@@ -115,6 +137,44 @@ class DatePickerInput extends React.Component {
     }
   }
 
+  static getDerivedStateFromProps(props, state) {
+    const tempDate = DateUtil.createSafeDate(props.value);
+    if (tempDate && props.value !== state.prevDate) {
+      return {
+        day: props.value.substr(8, 2),
+        month: props.value.substr(5, 2),
+        year: props.value.substr(0, 4),
+        prevDate: props.value,
+      };
+    }
+    return null;
+  }
+
+  moveFocusOnChange(inputValue, type, dateVariant) {
+    // Move focus based on input selected and date variant.
+    if (inputValue.length === 2) {
```

Comment:
Not sure if we want to make the behavior for the auto focus while entering value to match the time-input. For the time-input if you enter the first value between 3-9 for the hour, it will prepend a 0 and then auto focus to the next input. It does not require two digits since an hour that starts with 3 to 9 is not acceptable. See https://github.com/cerner/terra-framework/blob/master/packages/terra-time-input/src/TimeInput.jsx#L335-L351

cc/ @neilpfeiffer

Reply:
Not having the auto focus behavior would actually make a lot of this significantly easier, and I could trim out quite a bit. It makes sense not to have it with regard to date formats, but part of the original issue was about inserting slashes, so I tried to implement it here. @neilpfeiffer if it isn't needed, let me know!

Reply:
Sorry for the confusion on my previous comment. What I intended to say is I think we do want the auto focus but I wonder if we want to auto prepend a 0 to the entered value similar to how the time-input works. 

For example,  if the user enters "3" for the month, this would imply that the month is "03" since a two digit month that starts with "3" is invalid. In this [example](https://engineering.cerner.com/terra-framework/tests/terra-time-input/time-input/time-input/default), as soon as you enter a value from 3-9, it will prepend a 0 and auto focus to the next field since you can't enter an hour greater than 24.

Reply:
Understood! Yeah, if that is desired, I can definitely throw that in.